### DISCUSSION TITLE
Snap existing polygons without user interaction

### DISCUSSION THREAD

**Discussion by bryanschoot** [geoman-io/leaflet-geoman#923] [Questions & Help]
Is it possible to snap existing polygons to one another without any user interaction?

**Comment by Falke-Design** [comment]
Whats the usecase? What do you want to do?

**Reply by bryanschoot** [reply]
I have multiple polygons initiated using leaflet `addTo(map)` between these polygon is spacing meaning they are not connected. I was wondering if it is possible to connect the polygons with Leaflet-Geoman without any user interaction on the map. 

Using snappable.

**Reply by Falke-Design** [reply]
It is not directly possible with Geoman. But you can use this function:

```
function moveToLatLng(toLatLng, fromLatlng, layer) {
    // delta coords (how far to drag)
    const deltaLatLng = {
      lat: toLatLng.lat - fromLatlng.lat,
      lng: toLatLng.lng - fromLatlng.lng,
    };

    // move the coordinates by the delta
    const moveCoords = coords =>
      // alter the coordinates
      coords.map(currentLatLng => {
        if (Array.isArray(currentLatLng)) {
          // do this recursively as coords might be nested
          return moveCoords(currentLatLng);
        }

        // move the coord and return it
        return {
          lat: currentLatLng.lat + deltaLatLng.lat,
          lng: currentLatLng.lng + deltaLatLng.lng,
        };
      });


    // create the new coordinates array
    const newCoords = moveCoords(layer.getLatLngs());
    // set new coordinates and redraw
    layer.setLatLngs(newCoords);
  }
```
`fromLatlng` is one point of the layer that should be dragged
`toLatLng` is one point of the layer to which the other layer should dragged
`layer` layer that should be dragged
